Dan.Core/Services/AvailableEvidenceCodesService.cs

```diff
@@ -25,8 +26,8 @@ public class AvailableEvidenceCodesService : IAvailableEvidenceCodesService
 
     private List<EvidenceCode> _memoryCache = new();
     private DateTime _updateMemoryCache = DateTime.MinValue;
-    private readonly SemaphoreSlim _semaphoreForceRefresh = new(1, 1);
-    private readonly SemaphoreSlim _semaphore = new(1, 1);
+    private readonly AsyncNonKeyedLocker _semaphoreForceRefresh = new(1);
+    private readonly AsyncNonKeyedLocker _semaphore = new(1);
```

Comment:
How does AsyncNonKeyedLocker measure versus SemaphoreSlim? I don't mind the change, but I'm just not familiar with your library too much to know how they differ in terms of performance n such [:

Reply:
It's internally doing the same thing, using a semahoreslim, awaiting it, running the code it needs to run and finally release the semaphore. It's just syntactic sugar.